index.js

```diff
@@ -1016,41 +1020,63 @@ const createFocusTrap = function (elements, userOptions) {
       return this;
     },
 
+    _isManuallyPaused() {
+      return state.manuallyPaused;
+    },
+
+    _setPausedState(paused, options) {
+      state.paused = paused;
+      if (paused) {
+        const onPause = getOption(options, 'onPause');
+        const onPostPause = getOption(options, 'onPostPause');
+        onPause?.();
+
+        removeListeners();
+        updateObservedNodes();
+
+        onPostPause?.();
+      } else {
+        const onUnpause = getOption(options, 'onUnpause');
+        const onPostUnpause = getOption(options, 'onPostUnpause');
+
+        onUnpause?.();
+
+        updateTabbableNodes();
+        addListeners();
+        updateObservedNodes();
+
+        onPostUnpause?.();
+      }
```

Comment:
Wrapping this in `if (paused !== state.paused) { ... }` would preserve the old behavior more closely in terms of code checks. Right now, on auto-pause, if already paused, it would still remove listeners (again) and update observed nodes (again).

Not sure if that ever happens (auto-pausing an already paused trap) but just to cover the "what if", we should only actually do the behavior if there's an actual change in the state.

Reply:
Done